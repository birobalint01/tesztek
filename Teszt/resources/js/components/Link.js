import {useState} from "react";

const STATUSES = {
    HOVERED: 'hovered',
    NORMAL: 'normal'
}

export default function Link(props) {
    const [status, setStatus] = useState();

    const onMouseEnter = () => {
        setStatus(STATUSES.HOVERED)
    }

    const onMouseLeave = () => {
        setStatus(STATUSES.NORMAL)
    }

    const {
        page,
        children
    } = props;

    return (
      <a
        className={status}
        href={page || '#'}
        onMouseEnter={onMouseEnter}
        onMouseLeave={onMouseLeave}
      >
          {children}
      </a>
    );
}
