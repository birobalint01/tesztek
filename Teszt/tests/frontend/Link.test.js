import renderer from 'react-test-renderer';
import Link from '../../resources/js/components/Link';
;

it('change class when hovered', () => {
    const component = renderer.create(
    
        <Link></Link>
    
    );
    
    let tree = component.toJSON();
    
    expect(tree).toMatchSnapshot();

    renderer.act(() => {
    
        tree.props.onMouseEnter();
    
    })

    tree = component.toJSON();
   
    expect(tree).toMatchSnapshot();

    renderer.act(() => {
   
        tree.props.onMouseLeave();
    })

    tree = component.toJSON();
    
    expect(tree).toMatchSnapshot();
})