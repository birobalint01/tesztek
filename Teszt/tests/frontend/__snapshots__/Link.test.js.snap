// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`change class when hovered 1`] = `
<a
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
/>
`;

exports[`change class when hovered 2`] = `
<a
  className="hovered"
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
/>
`;

exports[`change class when hovered 3`] = `
<a
  className="normal"
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
/>
`;
